homeassistant/components/luftdaten/__init__.py

```diff
@@ -172,12 +172,9 @@ def _async_fixup_sensor_id(hass, config_entry, sensor_id):
     )
     remove_listener()
 
-    for component in ("sensor",):
-        await hass.config_entries.async_forward_entry_unload(config_entry, component)
-
     hass.data[DOMAIN][DATA_LUFTDATEN_CLIENT].pop(config_entry.entry_id)
 
-    return True
+    return await hass.config_entries.async_forward_entry_unload(config_entry, "sensor")
```

Comment:
Are we sure it's safe to change the order of steps here, ie that we forward to the platform after popping the client from hass.data?

Reply:
I think it's safe:
- In the `sensor.py`, there is only setup functions, no unload ones
- except here, the `DATA_LUFTDATEN_CLIENT` key is used only on setup functions

Reply:
Ok. :+1: